tests/test_Sequence.cc

```diff
@@ -3911,3 +3911,54 @@ TEST_CASE("Sequence: Check setter/getter function for step setup script", "[Sequ
         REQUIRE(ctx.step_setup_script == script);
     }
 }
+
+TEST_CASE("Sequence: sequence timeout", "[Sequence]")
+{
+    Step step{Step::type_action};
+    step.set_script("sleep(0.2)");
+
+    Sequence seq{"test_sequence"};
+    seq.push_back(std::move(step));
+
+    seq.set_timeout(100ms);
+
+    Context ctx;
+    auto maybe_error = seq.execute(ctx, nullptr);
+
+    REQUIRE(maybe_error.has_value() == true);
+    REQUIRE_THAT(maybe_error->what(), Contains("Timeout: Sequence"));
+}
+
+TEST_CASE("Sequence: sequence timeout is lower than step timeout", "[Sequence]")
+{
+    // TODO (Marcus, 13.02.2022):
```

Comment:
We need more improvement for checking sequence timeout. The current implementation is not enough.

I see two scenarios which would solve the problem as mentioned as a comment in the test case:

1. use another thread when staring a sequence that handles the timeout. But I can imagine that it can be tricky to abort the sequence.
2. use `Context` with two additional fields: one for start time and one for timeout. Both can be checked in the Lua hook `hook_check_timeout_and_termination_request`

But may there are better solutions than the above ones.

Reply:
We go now with the second approach. In `Context` there is a variable of type `TimeoutTrigger` that evaluates a timeout during executing a Lua script. Nevertheless there must also be a sequence timeout check when no Lua script is executed at all.

**My proposal:** check it also in `Sequence::execute_range` (as previously implemented). But I don't know how to implement a test case for that. Any suggestions?